README.md

```diff
@@ -89,7 +89,10 @@ limited by the capacity of a single physical machine, lest we sacrifice the
 property of complete auditability.  The Lightning Network [\[7\]][7] can help
 scale Bitcoin transaction volume by leaving some transactions off the ledger
 completely and is well suited for micropayments and privacy preserving payment
-rails, but may not be suitable for more generalized scaling needs.
+rails, and it is suitable for more generalized scaling needs.
+<!---
+state-channels allow for generalized smart contracts in the lightning network
+--->
```

Comment:
you could use the lightning state network to send an asset linked to the price of gold from one chain to another. bitcoin and litecoin don't off state-channels, they only have payment channels.

name-reg usually wouldn't go into channels, although it could.
If we are playing a game of poker, there could be 5 players who are matched to the 5 names "player 1", "player 2" ... "player 5".  that sort of name-reg would make sense in channels.
But domain names are much less likely, but technically we could do them too.
state-channels are a turing complete platform upon which we can implement consensus mechanisms like tendermint or flying fox or POW bitcoin. 
They are like sharding in that they are a scalability tool.

For domain registration: each participant makes a channel with a central node.
All the channels simultaneously load the same state which links a pubkey to a username for each person.
The channel can be updated with transactions that spend tokens from one user to another by username, you don't have to write the pubkey in the transaction, since the pubkey is written in the channel state.